File: frontend/src/model/Kpi.ts
import {TargetForKpi} from "./TargetForKpi";
import {MonthValuePair} from "./MonthValuePair";

export type Kpi = {
    id: string,
    name: string,
    ownedBy: string
    values: MonthValuePair[],
    targetForKpi: TargetForKpi,
    currentAverageValue: number,
}


export type NewKpi = {
    name: string,
    ownedBy: string,
    targetForKpi: TargetForKpi,
}

File: backend/src/main/java/capstone/kpiboard/model/kpi/Kpi.java
package capstone.kpiboard.model.kpi;

import org.springframework.data.annotation.Id;

import javax.validation.constraints.NotNull;
import java.util.List;

public record Kpi(
        @Id
        String id,
        @NotNull(message = "Bitte einen Namen eingeben")
        String name,
        @NotNull(message = "Bitte eine Verantwortlichkeit eingeben")
        String ownedBy,
        List<MonthValuePair> values,
        @NotNull(message = "Bitte ein Zielwert eingeben")
        TargetForKpi targetForKpi,
        double currentAverageValue
) {
}
